### PR TITLE
feat: expose Claude MCP server

### LINKED ISSUE #33: feat: experiment with Claude MCP
Claude recently announced Model Context Providers, it is also generally available in Zed, and likely soon, in other dev tools. Would be great to expose that React Native agent to these tools via that protocol. Shouldn't take too much (hopefully) with Claude AI SDK.

Work in progress!

### PR DESCRIPTION
**Pull request by grabbou**
Fixes #33 
Added Builder Bob to consumable package (for context, Bun doesn't produce declarations, and since it bundles everything into a single file, that becomes a challenge) 

----

This PR adds experimental Model Context Protocol Server that can be used with Claude, Zed and other compatible environments. 

This is just a first take, there are other features in MCP, including resources, prompts that are all useful and we will be exploring them later!

Going forward, we should unify Vercel AI SDK tools with this, potentially, moving entirely to MCP protocol and using that with Cali as MCP Client in Terminal.

### CODE REVIEW

**Comment by grabbou** [comment]
Note: This is experimental. Not all tools are obviously working, but we will get there soon!

**Comment by grabbou** [comment]
Going to merge and ship it @okwasniewski, but let's do a CR anyways and we will do a follow-up on Monday if necessary! 🙏

File: packages/mcp-server/src/index.ts [hunk 1]
```diff
@@ -1 +1,110 @@
-console.log('Cooming soon! (tm)')
+#!/usr/bin/env node
+
+import * as tools from '@cali/tools-react-native'
+import { Server } from '@modelcontextprotocol/sdk/server/index.js'
+import { StdioServerTransport } from '@modelcontextprotocol/sdk/server/stdio.js'
+import { CallToolRequestSchema, ListToolsRequestSchema } from '@modelcontextprotocol/sdk/types.js'
```
Comment (okwasniewski) [inline packages/mcp-server/src/index.ts:6]: Do these imports need .js extensions?

File: packages/mcp-server/src/index.ts [hunk 2]
```diff
@@ -1 +1,110 @@
...
+import zodToJsonSchema from 'zod-to-json-schema'
+
+const server = new Server(
+  {
+    name: '@cali/mcp-server',
+    version: '0.1.0',
```
Comment (okwasniewski) [inline packages/mcp-server/src/index.ts:12]: nit: maybe let's read this from package.json?

**Pull request merged**